src/java.base/share/classes/java/lang/Process.java

```diff
@@ -161,6 +169,193 @@ public Process() {}
+     * <p>If the standard input of the process has been redirected using
+     * {@link ProcessBuilder#redirectInput(Redirect)
+     * ProcessBuilder.redirectInput} then this method will return a
+     * <a href="ProcessBuilder.html#redirect-input">null output writer</a>.
+     *
+     * @apiNote
+     * A {@linkplain BufferedWriter} writes characters, arrays of characters, and strings.
+     * Wrapping the {@link BufferedWriter} with a {@link PrintWriter} provides
+     * efficient buffering and formatting of primitives and objects as well as support
+     * for auto-flush on line endings.
+     * <p>
+     * Use {@link #getOutputStream} and {@link #outputWriter} with extreme care.
+     * Output to the BufferedWriter may be held in the buffer until
+     * {@linkplain BufferedWriter#flush flush} is called.
+     *
+     * @return a BufferedWriter to the standard input of the process using the charset
+     *          for the {@code native.encoding} system property
+     * @since 17
+     */
+    public BufferedWriter outputWriter() {
+        return outputWriter(CharsetHolder.nativeCharset());
+    }
+
+    /**
+     * Returns a {@code BufferedWriter} connected to the normal input of the process using a Charset.
+     * Writes text to a character-output stream, buffering characters so as to provide
+     * for the efficient writing of single characters, arrays, and strings.
+     *
+     * <p>Characters written by the writer are encoded to bytes using {@link OutputStreamWriter}
+     * and the {@link Charset}
+     * piped into the standard input of the process represented by this {@code Process} object.
+     * Call the {@link BufferedWriter#flush()} method to flush buffered output to the process.
+     *
+     * <p>If the standard input of the process has been redirected using
+     * {@link ProcessBuilder#redirectInput(Redirect)
+     * ProcessBuilder.redirectInput} then this method will return a
+     * <a href="ProcessBuilder.html#redirect-input">null output writer</a>.
+     *
+     * @apiNote
+     * A {@linkplain BufferedWriter} writes characters, arrays of characters, and strings.
+     * Wrapping the {@link BufferedWriter} with a {@link PrintWriter} provides
+     * efficient buffering and formatting of primitives and objects as well as support
+     * for auto-flush on line endings.
+     * <p>
+     * Use {@link #getOutputStream} and {@link #outputWriter} with extreme care.
+     * Output to the BufferedWriter may be held in the buffer until
+     * {@linkplain BufferedWriter#flush flush} is called.
+     *
+     * @param charset the {@code Charset} to encode characters to bytes, not null
+     * @return a BufferedWriter to the standard input of the process using the {@code charset}
+     * @throws NullPointerException if the {@code charset} is {@code null}
+     * @since 17
+     */
+    public BufferedWriter outputWriter(Charset charset) {
+        return new BufferedWriter(new OutputStreamWriter(getOutputStream(), charset));
```

Comment:
I'd suggest calling Objects.requireNonNull(charset) first thing in this method instead of relying on the InputStreamWriter constructor to throw the NPE.